src/main/java/rx/schedulers/JavaFxScheduler.java

```diff
@@ -46,86 +48,167 @@ public static JavaFxScheduler getInstance() {
+            long delay = unit.toMillis(delayTime);
+            assertThatTheDelayIsValidForTheJavaFxTimer(delay);
 
-            final long delay = unit.toMillis(max(delayTime, 0));
-            final Timeline timeline = new Timeline(new KeyFrame(Duration.millis(delay), new EventHandler<ActionEvent>() {
+            class DualAction implements EventHandler<ActionEvent>, Subscription, Runnable {
+
+                private Timeline timeline;
+
+                final SerialSubscription subs = new SerialSubscription();
+                boolean nonDelayed;
+
+                private void setTimer(Timeline timeline) {
+                    this.timeline = timeline;
+                }
 
                 @Override
                 public void handle(ActionEvent event) {
-                    if (innerSubscription.isUnsubscribed() || s.isUnsubscribed()) {
-                        return;
-                    }
-                    action.call();
-                    innerSubscription.remove(s);
+                    run();
                 }
-            }));
-
-            timeline.setCycleCount(1);
-            timeline.play();
 
-            innerSubscription.add(s);
+                @Override
+                public void run() {
+                    if (nonDelayed) {
+                        try {
+                            if (tracking.isUnsubscribed() || isUnsubscribed()) {
+                                return;
+                            }
+                            action.call();
+                        } finally {
+                            subs.unsubscribe();
+                        }
+                    } else {
+                        timeline.stop();
+                        timeline = null;
+                        nonDelayed = true;
+                        trampoline(this);
+                    }
+                }
 
-            // wrap for returning so it also removes it from the 'innerSubscription'
-            return Subscriptions.create(new Action0() {
+                @Override
+                public boolean isUnsubscribed() {
+                    return subs.isUnsubscribed();
+                }
 
                 @Override
-                public void call() {
-                    timeline.stop();
-                    s.unsubscribe();
-                    innerSubscription.remove(s);
+                public void unsubscribe() {
+                    subs.unsubscribe();
+                }
+                public void set(Subscription s) {
+                    subs.set(s);
                 }
+            }
 
-            });
+            final DualAction executeOnce = new DualAction();
+            tracking.add(executeOnce);
+
+            //final Timeline timer = new Timeline((int) delay, executeOnce);
+            final Timeline timer = new Timeline(new KeyFrame(Duration.millis(delay)));
+
+            executeOnce.setTimer(timer);
+            timer.play();
```

Comment:
Does the Timeline execute its action on the JavaFX thread? If not, then the `trampoline()`in its current form won't work.